Insolvency.CalculationsEngine.Redundancy.API/Infrastructure/Middlewares/Validators/ArrearsOfPayCalculationRequestValidator.cs

```diff
@@ -58,6 +60,32 @@ public ArrearsOfPayCalculationRequestValidator()
             RuleFor(req => req.PayDay).Must(CommonValidation.BeValidPayDay)
                 .WithMessage(
                     $"'Pay day' is not valid correct values are [0 = Sunday, 1 = Mon, 2 = Tues, 3 = Wed, 4 = Thurs, 5 = Fri, 6 = Sat]");
+
+            RuleFor(req => req)
+            .Must(WorkingDaysInClaimPresent)
+            .WithMessage($"No Working Days in Claim");
+        }
+
+        private bool WorkingDaysInClaimPresent(ArrearsOfPayCalculationRequestModel appa)
+        {
+            int numDays = 0;
+            DateTime start = appa.UnpaidPeriodFrom;
+            DateTime end = appa.UnpaidPeriodTo;
+
+            if (end >= start)
+            {
+                if (appa.ShiftPattern != null)
+                {
+                    string weekDayNames = appa.ShiftPattern.GetShiftDayNames().Result;
+
+                    for (var date = start; date <= end; date = date.AddDays(1))
+                    {
+                        if (weekDayNames.Contains(date.DayOfWeek.ToString()))
+                            numDays++;
```

Comment:
Rather than counting the shift days, could this just return true when the first one is found.

Reply:
Thanks for reviewing Brian, yes it definitely good idea to return true